File: frontend/src/types/analytics.ts
export interface TraineeSessionAnalyticsDTO {
  session_id: number
  case_id: number
  case_title: string
  empathy_score: number
  communication_score: number
  clinical_score: number
  spikes_completion_score: number
  spikes_coverage_percent: number
  overall_score?: number | null
  duration_seconds: number
  created_at: string
  eo_addressed_rate?: number | null
  spikes_stages_covered?: string[] | null
}

export interface TraineeSessionAnalytics {
  sessionId: number
  caseId: number
  caseTitle: string
  empathyScore: number
  communicationScore: number
  clinicalScore: number
  spikesCompletionScore: number
  spikesCoveragePercent: number
  /** Backend-provided overall score when available; otherwise computed on the client. */
  overallScore?: number
  durationSeconds: number
  createdAt: string
  eoAddressedRate?: number
  /** SPIKES stage codes present in the session (e.g. S, P, I), when available from feedback. */
  spikesStagesCovered?: string[]
}

File: backend/src/domain/models/analytics.py
"""Trainee analytics schemas."""

from typing import Optional

from pydantic import BaseModel

from core.time import UTCDateTime


class TraineeSessionAnalytics(BaseModel):
    """Per-session analytics row for the authenticated trainee."""

    session_id: int
    case_id: int
    case_title: str
    empathy_score: float
    communication_score: float
    clinical_score: float
    spikes_completion_score: float
    spikes_coverage_percent: float
    duration_seconds: int
    created_at: UTCDateTime
    eo_addressed_rate: Optional[float] = None
    spikes_stages_covered: Optional[list[str]] = None

